vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -34,17 +42,52 @@
 
     @Autowired
     private SequenceRepository sequenceRepo;
-    
+
+    @Autowired
+    private BranchingPointRepository bpointRepo;
+
+    @Autowired
+    private ChoiceRepository choiceRepo;
+
+    @Autowired
+    private ChoiceFactory choiceFactory;
+
     private final Logger logger = LoggerFactory.getLogger(getClass());
-    
-    @Override
-    public ISlide createSlide(String moduleId, SlideForm slideForm) {
-        IModule module = moduleManager.getModule(moduleId);
-        ISlide slide = slideFactory.createSlide(module, slideForm);
-        slideRepo.save((Slide) slide);
-        return slide;
+
+    @Override
+    public ISlide createSlide(IModule module, SlideForm slideForm, SlideType type) {
+        ISlide slide = slideFactory.createSlide(module, slideForm, type);     
+        return slideRepo.save((Slide) slide);
+    }
+
+    @Override
+    public IBranchingPoint createBranchingPoint(IModule module, SlideForm slideForm, SlideType type) {
+        ISlide branchingPoint = slideFactory.createSlide(module, slideForm, type);                 
+        return bpointRepo.save((BranchingPoint) branchingPoint);
     }
 
+    @Override
+    public void updateBranchingPoint(IBranchingPoint branchingPoint, List<String> editedChoices) {
+        List<IChoice> existingChoices=branchingPoint.getChoices();
+        List<String> existingChoiceSequences=existingChoices.stream().map(choiceSequence -> choiceSequence.getSequence().getId()).collect(Collectors.toList());
+        List<String> deletedChoices = (List<String>) CollectionUtils.subtract(existingChoiceSequences, editedChoices);
+        List<String> addedChoices = (List<String>) CollectionUtils.subtract(editedChoices,existingChoiceSequences);
+        List<IChoice> newlyAddedChoices = choiceFactory.createChoices(addedChoices);

```

Comment:
it would be a lot easier to read if you call lists that contain ids "...Ids" and lists that contain choices "...Choices"